src/Syntax/Toplevel.hs

```diff
@@ -30,6 +30,7 @@ data Toplevel p
   = LetStmt TopAccess [Binding p]
   | ForeignVal TopAccess (Var p) Text (Type p) (Ann p)
   | TypeDecl TopAccess (Var p) [TyConArg p] (Maybe [Constructor p]) (Ann p)
+  | TySymDecl TopAccess (Var p) [TyConArg p] (Type p) (Ann p)
```

Comment:
Would it be better to have `TypeDecl` take something like `Empty | Alias (Type p) | Variant [Constructor p]`? Dunno if that'll actually simplify things or not.

Reply:
I don't think this will simplify things because of how different kinding is for type declarations (always returns `*`, constructors need to have a kind ending in `*`, etc) vs type aliases (any return kind)